docs/examples/my-fuel-dapp/frontend/src/App.tsx

```diff
@@ -24,30 +24,45 @@ const contract = ContractAbi__factory.connect(CONTRACT_ID, wallet);
 
 function App() {
   const [counter, setCounter] = useState(0);
+  const [balance, setBalance] = useState(0); 
 
   useEffect(() => {
     async function main() {
       // Executes the counter function to query the current contract state
       // the `.get()` is read-only, because of this it don't expand coins.
-      const { value } = await contract.functions.counter().get();
-      setCounter(Number(value));
+        const { value } = await contract.functions.counter().get();
+        setCounter(Number(value));
+        setBalance(Number(value));
     }
     main();
   }, []);
 
+  useEffect(() => {
+    let active = true;
+    async function fetch(){
+      if (active) {
+        // Creates a transactions to call the increment function passing the amount
+        // we want to increment, because it creates a TX and updates the contract state
+        // this requires the wallet to have enough coins to cover the costs and also
+        // to sign the Transaction
+        const { value } = await contract.functions.increment(1).call();
+        setBalance(Number(value));
+      }
+    }
+    fetch();
+    return () => {
+      active = false;
+    }
+  }, [counter]);
+
   async function increment() {
-    // Creates a transactions to call the increment function passing the amount
-    // we want to increment, because it creates a TX and updates the contract state
-    // this requires the wallet to have enough coins to cover the costs and also
-    // to sign the Transaction
-    const { value } = await contract.functions.increment(1).call();
-    setCounter(Number(value));
+    setCounter(balance + 1);
```

Comment:
I think we should have a loading state also, then won't need to have balance at all.

1- you can create loading: `const [loading, setLoading] = useState(false); `
2- this `increment` function can be created with useCallback, and listen to `loading`
3- in the start of `increment` function: `if (loading) return`
4- before you execute contract call: `await setLoading(true)`
4- after you execute contract call `await setLoading(false)`

then will never be called twice

Reply:
Other strategy would be to `debounce` or `throttle` the click function

Reply:
Thanks! @LuizAsFight that's way more clever rather than the cleanup hook. I've even switched the code back to what it was and simply added a loading state and that's all.